test/CadlRanchProjects/type/property/nullable/src/Generated/Models/CollectionsByteProperty.Serialization.cs

```diff
@@ -32,6 +32,11 @@ internal static CollectionsByteProperty DeserializeCollectionsByteProperty(JsonE
                 }
                 if (property.NameEquals("nullableProperty"u8))
                 {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
```

Comment:
I believe we should not do this.
We will get into this if when we get this json in response:
```
{ "nullableProperty": null }
```
If we want a model to serialize  into the json above, we will need to ensure the value of this collection is `null`. And we should do the same here when it is in the response, this makes the serialization and deserialization round trip.

Here are the model definition:
```csharp
public partial class Foo
{
    public Foo(IEnumerable<string> requiredNullableList)
    {
		RequiredNullableList = requiredNullableList?.ToList();
	}

	public IList<string> RequiredNullableList { get; set; }
}
```
If we have this code:
```csharp
var model = new Foo(null);
```
it will serialize into:
```json
{ "requiredNullableList": null }
```
And if you deserialize from the above json, you will get an instance of Foo, and its `RequiredNullableList` property is `null` again, something like:
```csharp
var model = new Foo(null);
var json = Serialize(model); // json is `{ "requiredNullableList": null }`
var deserializedModel = Deserialize(json);
Console.WriteLine(model.RequiredNullableList == deserializedModel.RequiredNullableList); // true
```
Same thing applies to optional list. If we have:
```csharp
var model = new Foo();
model.OptionalNullableList = null;
```
it serializes into
```json
{ "optionalNullableList": null }
``` 
and when we deserialize above json, we get an instance of `Foo` whose `OptionalNullableList` property is null.
If we have:
```csharp
var model = new Foo();
```
it serializes into nothing:
```json
{ }
```
and when we deserialize above json, we get an instance of `Foo` whose `OptionalNullableList` property is the ChangeTrackingList with `IsUndefined = true`.
I will add some more test cases in `Models-TypeSpec` to demo these.

Reply:
We resolved this in offline discussion